### ISSUE TITLE
BC6H decoder seems to produce broken results

### ISSUE THREAD

**Issue by aras-p** [elasota/ConvectionKernels#6]
I'm trying to use `cvtt::Internal::BC6HComputer::UnpackOne` directly, to decode a BC6H (unsigned) image block by block. With various other decompression libraries (DirectXTex and others) everything seems to work, but cvtt seems to produce "something entirely else". Happens with all BC6H files I tried so far, on both Mac (clang 13) & Windows (vs2022).

Attached the source file (`IndoorHDRI001.dds`), what is the expected decoding result (with DirectXTex in this case), `IndoorHDRI001-dxtex.hdr` and what I get with cvtt, `IndoorHDRI001-convection.hdr`. Github does not allow attaching dds/hdr directly, so they are all in a zip file.

[IndoorHDRI001.zip](https://github.com/elasota/ConvectionKernels/files/8959933/IndoorHDRI001.zip)

My usage is basically like this:
```
cvtt::PixelBlockF16 rgba;
for (int i = 0; i < height; i += 4)
{
    for (int j = 0; j < width; j += 4)
    {
        cvtt::Internal::BC6HComputer::UnpackOne(rgba, src, format == DDSFormat::BC6HS);
        src += 16;
        for (int c = 0; c < 16; ++c) {
            float* dst_f = (float*)(dst + (i*width+j)*12 + width*(c/4)*12 + (c&3)*12);
            dst_f[0] = half_to_float_fast5(rgba.m_pixels[c][0]);
            dst_f[1] = half_to_float_fast5(rgba.m_pixels[c][1]);
            dst_f[2] = half_to_float_fast5(rgba.m_pixels[c][2]);
        }
    }
}
```

**Comment by elasota** [comment]
Looks like something went wrong with the bit alignment in the header decoder.  It's decoding as if the header block is aligned from the block start, instead of from after the mode bits, so all of the bits are 2 or 5 bits out of position.

**Comment by elasota** [comment]
Fixed by 350416daa4e98f1c17ffc273b134d0120a2ef230
